### ISSUE TITLE
Unsuccessfully adding or editing specimens

### ISSUE THREAD

**Issue by mattcgo** [spawaskar-cora/cora-docs#438]
Hello,
I have been unsuccessful in adding or editing specimens. We've tried it across two accounts on two different computers and the problem is consistent. I was trying to edit Designator 503 in Accession 2019-134 and failing, but editing Designator 1 was being successful. I was also unsuccessful in adding new specimens or specimen groups. It was working earlier today but hasn't been since.
Thanks,
Matt

**Comment by SachinPawaskarUNO** [comment]
@mattcgo, @fedamann The logs suggest that you first tried to add/insert the following records with the Composite Key of as shown below.
2019-134,AGRS,ROK,503
2019-134,AGRS,ROK,504
2019-134,AGRS,ROK,509
2019-134,AGRS,ROK,513
These failed because Specimen records with these as Composite Key already exist. 

TheI recommended workflow is to first delete the Specimen record that you made the error on and then add/insert the new record with the correction.
Please Note you do that have the authorization level to delete specimen or DNA records, this will have to be done by your project lead. I would recommend you and your project lead connect with @clegarde or @cbrown311 as they have done on their project and might be able explain/walk you through the process.

**Comment by SachinPawaskarUNO** [comment]
@mattcgo, @fedamann One more things I forgot to mention is that authorization level to delete records (in this case Specimen) is available to the project lead and org Admins.

**Comment by fedamann** [comment]
@veronicak7 is authorized to delete records for KWP

**Comment by veronicak7** [comment]
Got it, thank you for the clarification!

**Comment by mattcgo** [comment]
@SachinPawaskarUNO thank you for looking into this. However, if I look at the actual 2019-134 accession, I see there are only
2019-134,AGRS,ROK,1
2019-134,AGRS,ROK,2
2019-134,AGRS,ROK,500
2019-134,AGRS,ROK,501
2019-134,AGRS,ROK,502
2019-134,Namjong-gu,ROK,503

How can the composite keys for 504, 509, and 513 already exist? If they do, I cannot delete them on my end as I cannot see them. Also, I had successfully deleted several other specimens from this accession previously.
As for 503, the "Namjong-gu" component is a typo but I was unsuccessful in editing it to the correct "AGRS" provenience. I just tried what you suggested and deleted 503, which successfully deleted but I am still unsuccessful in adding it back in. It seems some of the designators in this accession are frozen in limbo, and although I cannot see them they are created somewhere and so I cannot add or edit them. May I suggest we just delete everything about 2019-134 and I start from scratch?

I am using Safari on Mac iOS 10.14.4, and we also still saw these problems using @veronicak7 's DPAA Windows computer.

Matt

**Comment by SachinPawaskarUNO** [comment]
@mattcgo and @fedamann, Here is what the data is telling me. The following records exists for Accession Number/Provenance1/Provenance2: 2019-134, ARGS, ROK, All were created by you. Only one with designator 1 was last updated by @veronicak7. I'm also seeing that you deleted the records starting with designator 503 through 517. These were created as part of Maxillary Dentition bone group. 

2019-134,AGRS,ROK,1
2019-134,AGRS,ROK,2
2019-134,AGRS,ROK,500
2019-134,AGRS,ROK,501
2019-134,AGRS,ROK,502
2019-134,AGRS,ROK,503
2019-134,AGRS,ROK,504
2019-134,AGRS,ROK,505
2019-134,AGRS,ROK,506
2019-134,AGRS,ROK,507
2019-134,AGRS,ROK,508
2019-134,AGRS,ROK,509
2019-134,AGRS,ROK,510
2019-134,AGRS,ROK,511
2019-134,AGRS,ROK,512
2019-134,AGRS,ROK,513
2019-134,AGRS,ROK,514
2019-134,AGRS,ROK,515
2019-134,AGRS,ROK,516
2019-134,AGRS,ROK,517

What I think is happening is that the records are being soft deleted and still in the database for auditing purposes. For now I will go ahead with what you have suggested and remove all records in Accession 2019-134. I will follow up with @fedamann and see how we want to handle the auditing process in the future.

**Comment by mattcgo** [comment]
@SachinPawaskarUNO thank you. I started adding specimens into this accession again. I tried adding a bone group for the cervical vertebrae beginning with Designator 504, and it gave me a message that it was successfully started. However, it does not show up in the accession. I suspect this is related to our previous problem. Are you able to see on your backend that the specimens 504-510 were added but they don't appear in the accession?

**Comment by mattcgo** [comment]
> @SachinPawaskarUNO thank you. I started adding specimens into this accession again. I tried adding a bone group for the cervical vertebrae beginning with Designator 504, and it gave me a message that it was successfully started. However, it does not show up in the accession. I suspect this is related to our previous problem. Are you able to see on your backend that the specimens 504-510 were added but they don't appear in the accession?

Nevermind @SachinPawaskarUNO , I think there was a delay because the specimens showed up in the accession 20 minutes later.

**Comment by SachinPawaskarUNO** [comment]
@mattcgo That makes sense, because when you create Specimens via Bone Groups, they are created via a queued job. Once the job in completed you will receive a notification.

**Comment by mattcgo** [comment]
@SachinPawaskarUNO I've run into this problem again where I am able to successfully delete specimens (which I shouldn't be able to do?), but then cannot reuse the designator.
I deleted: 2019-099,AGRS,ROK,521,Rib 1 but I cannot put in 2019-099,AGRS,ROK,521,Sacrum.

**Comment by SachinPawaskarUNO** [comment]
@mattcgo I have purged the record. So you should be able to add a new one.
@fedamann Can you check with Matt and see the reasons for these errors. I'm curious if there is something we can do to mitigate these data entry errors. We should also discuss the audit/purge process.

**Comment by fedamann** [comment]
@mattcgo I'll call you tomorrow to discuss. Are these data entry errors resulting from the bone group inventory process that produces a record that duplicates some other in your legacy data sheet? Can send me the data sheet your working from or direct my attention to where it sits on the network.

**Comment by mattcgo** [comment]
@fedamann @SachinPawaskarUNO I have talked to other KWIP analysts on the project here using CoRA. We now know that we are all experiencing the problem of deleting a specimen but not being able to reuse the designator. @fedamann There is no legacy data sheet for the K208. I have been manually writing down the elements from old CHR reports in the records room and the KMC.

I am attaching here a small portion of elements from two accessions in the K208 using the template @SachinPawaskarUNO had provided @veronicak7 last year. I was hoping to use this spreadsheet to test whether the data imports well into CoRA. Moreover, if it does import well, may I please ask that a template that also accommodates the DNA results also be created?
[se_sample_template.xlsx](https://github.com/spawaskar-cora/cora-docs/files/3119352/se_sample_template.xlsx)

PS. @fedamann my DPAA extension is x3803. I do not have network access yet, and won't be granted access for quite some time.

**Comment by fedamann** [comment]
@mattcgo this is very helpful, thank you!  And based on earlier comments I assume the numeric data in the Designator column is the exact specimen identifier pulled from the CHR reports per specimen?

**Comment by mattcgo** [comment]
> @mattcgo this is very helpful, thank you! And based on earlier comments I assume the numeric data in the Designator column is the exact specimen identifier pulled from the CHR reports per specimen?

That is correct, but only if a CHR report was actually written. Some of the accessions predate CHR and therefore I assigned designators myself.

**Comment by amysears1022** [comment]
Hi @SachinPawaskarUNO and @fedamann 

I have had the same issue in CoRA with a case I'm attempting to enter, 2019-051. I mistakenly created two bone groups, and had to delete the second bone group. As such, the designators 2019-051-AGRS-ROK-539 and up cannot currently be used. I understand that by me deleting those entires, I merely soft deleted the elements, however @veronicak7 attempted to go in and delete the elements as well, and was unable to. As previously stated, she should have the authorization to delete specimens, however they do not show up for her.

**Comment by SachinPawaskarUNO** [comment]
@amysears1022 Can you tell me which two bone groups you created and which one needs to be deleted? I need the AN, P1, P2 and Designator ranges for each.

**Comment by amysears1022** [comment]
The bone group I wanted, a grouping of nine hand bones, which I have not deleted, is:
2019-051-AGRS-ROK-527
 2019-051-AGRS-ROK-528
2019-051-AGRS-ROK-529
2019-051-AGRS-ROK-530
2019-051-AGRS-ROK-531
2019-051-AGRS-ROK-532
2019-051-AGRS-ROK-533
2019-051-AGRS-ROK-534
2019-051-AGRS-ROK-535


The specimens I need deleted are:
2019-051-AGRS-ROK-539 and above (540, 541, 542, etc). I do not remember what the last number in the grouping was, either 546 or 555). They were duplicated elements.

**Comment by SachinPawaskarUNO** [comment]
@amysears1022 I have force deleted the requested Specimen records. There were 17 records in all with 2019-051-AGRS-ROK and designators from 539 through 555. Let me know if that addressed your issue for now.

**Comment by amysears1022** [comment]
@SachinPawaskarUNO That worked, thanks!
